### PR TITLE
Use logrus instead of a custom logger

### PR DESCRIPTION
**Pull request by guillaumerose**
The RPC now takes a logrus instance as a parameter (both server and client side).

Changes to propagate in drivers:
Replace `"github.com/code-ready/machine/libmachine/log"` by `log "github.com/sirupsen/logrus"`

Changes to propagate in crc:
No more  setMachineLogging with log.SetOutWriter(logfile), log.SetErrWriter(logfile).

It change the output of crc as it unifies the logging.
When running `crc start --log-level debug`:

```
INFO Creating CodeReady Containers VM for OpenShift 4.6.9... 
DEBU Found binary path at /home/guillaumerose/.crc/bin/crc-driver-libvirt 
DEBU Launching plugin server for driver libvirt   
DEBU Plugin server listening at address 127.0.0.1:36849 
DEBU () Calling .GetVersion                       
DEBU Using API Version 1                          
DEBU () Calling .SetConfigRaw                     
DEBU () Calling .GetMachineName                   
DEBU (crc) Calling .GetMachineName                
DEBU (crc) Calling .DriverName                    
DEBU Running pre-create checks...                 
DEBU (crc) Calling .PreCreateCheck                
DEBU (crc) DBG | time="2021-01-19T15:40:50+01:00" level=debug msg="About to check libvirt version" 
DEBU (crc) DBG | time="2021-01-19T15:40:50+01:00" level=debug msg="Validating storage pool" 
DEBU (crc) Calling .GetConfigRaw                  
DEBU Creating machine...                          
DEBU (crc) Calling .Create                        
DEBU (crc) DBG | time="2021-01-19T15:40:50+01:00" level=debug msg="Preparing /home/guillaumerose/.crc/machines/crc/crc.qcow2 for machine use" 
DEBU (crc) DBG | time="2021-01-19T15:40:50+01:00" level=debug msg="image creation took 11.253505ms" 
DEBU (crc) DBG | time="2021-01-19T15:40:50+01:00" level=debug msg="Verifying executable bit set on /home/guillaumerose/.crc/machines/crc" 
DEBU (crc) DBG | time="2021-01-19T15:40:50+01:00" level=debug msg="Setting executable bit set on /home/guillaumerose/.crc/machines/crc" 
DEBU (crc) DBG | time="2021-01-19T15:40:50+01:00" level=debug msg="Verifying executable bit set on /home/guillaumerose/.crc/machines" 
DEBU (crc) DBG | time="2021-01-19T15:40:50+01:00" level=debug msg="Verifying executable bit set on /home/guillaumerose/.crc" 
DEBU (crc) DBG | time="2021-01-19T15:40:50+01:00" level=debug msg="Verifying executable bit set on /home/guillaumerose" 
DEBU (crc) DBG | time="2021-01-19T15:40:50+01:00" level=debug msg="Verifying executable bit set on /home" 
DEBU (crc) DBG | time="2021-01-19T15:40:50+01:00" level=debug msg="Defining VM..." 
``` 

In crc.log:
```
time="2021-01-19T15:33:30+01:00" level=info msg="Creating CodeReady Containers VM for OpenShift 4.6.9..."
time="2021-01-19T15:33:30+01:00" level=debug msg="Found binary path at /home/guillaumerose/.crc/bin/crc-driver-libvirt"
time="2021-01-19T15:33:30+01:00" level=debug msg="Launching plugin server for driver libvirt"
time="2021-01-19T15:33:30+01:00" level=debug msg="Plugin server listening at address 127.0.0.1:37101"
time="2021-01-19T15:33:30+01:00" level=debug msg="() Calling .GetVersion"
time="2021-01-19T15:33:30+01:00" level=debug msg="Using API Version 1"
time="2021-01-19T15:33:30+01:00" level=debug msg="() Calling .SetConfigRaw"
time="2021-01-19T15:33:30+01:00" level=debug msg="() Calling .GetMachineName"
time="2021-01-19T15:33:30+01:00" level=debug msg="(crc) Calling .GetMachineName"
time="2021-01-19T15:33:30+01:00" level=debug msg="(crc) Calling .DriverName"
time="2021-01-19T15:33:30+01:00" level=debug msg="Running pre-create checks..."
time="2021-01-19T15:33:30+01:00" level=debug msg="(crc) Calling .PreCreateCheck"
time="2021-01-19T15:33:30+01:00" level=debug msg="(crc) DBG | time=\"2021-01-19T15:33:30+01:00\" level=debug msg=\"About to check libvirt version\""
time="2021-01-19T15:33:30+01:00" level=debug msg="(crc) DBG | time=\"2021-01-19T15:33:30+01:00\" level=debug msg=\"Validating storage pool\""
time="2021-01-19T15:33:30+01:00" level=debug msg="(crc) Calling .GetConfigRaw"
time="2021-01-19T15:33:30+01:00" level=debug msg="Creating machine..."
time="2021-01-19T15:33:30+01:00" level=debug msg="(crc) Calling .Create"
time="2021-01-19T15:33:30+01:00" level=debug msg="(crc) DBG | time=\"2021-01-19T15:33:30+01:00\" level=debug msg=\"Preparing /home/guillaumerose/.crc/machines/crc/crc.qcow2 for machine use\""
time="2021-01-19T15:33:30+01:00" level=debug msg="(crc) DBG | time=\"2021-01-19T15:33:30+01:00\" level=debug msg=\"image creation took 11.766809ms\""
time="2021-01-19T15:33:30+01:00" level=debug msg="(crc) DBG | time=\"2021-01-19T15:33:30+01:00\" level=debug msg=\"Verifying executable bit set on /home/guillaumerose/.crc/machines/crc\""

```

The only drawback is that timestamp are duplicated in the log file.

### CODE REVIEW

File: libmachine/drivers/plugin/localbinary/plugin.go [hunk 1]
```diff
@@ -59,11 +43,18 @@ type McnBinaryExecutor interface {
...
-	PluginServer
-	PluginStreamer
+	// Get the address where the plugin server is listening.
+	Address() (string, error)
+
+	// Serve kicks off the plugin server.
+	Serve() error
+
+	// Close shuts down the initialized server.
+	Close() error
```
Comment (cfergeau) [inline libmachine/drivers/plugin/localbinary/plugin.go:53]: These changes seem unrelated?

File: libmachine/drivers/plugin/localbinary/plugin.go [hunk 2]
```diff
@@ -89,19 +81,21 @@ func driverPath(driverName string, binaryPath string) string {
...
 
-	log.Debugf("Found binary path at %s", binaryPath)
+	logger.Debugf("Found binary path at %s", binaryPath)
 
 	return &Plugin{
+		logger: logger,
 		stopCh: make(chan struct{}),
 		addrCh: make(chan string, 1),
 		Executor: &Executor{
+			logger:     logger,
```
Comment (cfergeau) [inline libmachine/drivers/plugin/localbinary/plugin.go:98]: I'm not a big fan of duplicating in each object something which is ultimately a shared resource and which will be the same for the whole process. This is just confusing when reading the code and trying to find out what creates this per-object thing, how this instance behaves differently from the others, ...

Comment (guillaumerose) [reply libmachine/drivers/plugin/localbinary/plugin.go:98]: The alternative is to use the `logrus.Info` etc directly. I can do and I think the output will be the same. Would it be better?

File: libmachine/drivers/plugin/localbinary/plugin.go (hunk 1 above)
Comment (guillaumerose) [reply libmachine/drivers/plugin/localbinary/plugin.go]: yes done.

File: libmachine/drivers/plugin/localbinary/plugin.go (hunk 2 above)
Comment (guillaumerose) [reply libmachine/drivers/plugin/localbinary/plugin.go]: I removed this logger and just used logrus everywhere. PTAL.

File: libmachine/drivers/rpc/client_driver.go [hunk 3]
```diff
@@ -36,7 +35,7 @@ func NewRPCClientDriverFactory() RPCClientDriverFactory {
 }
 
 type RPCClientDriver struct {
-	plugin          localbinary.DriverPlugin
+	plugin          io.Closer
```
Comment (cfergeau) [inline libmachine/drivers/rpc/client_driver.go:38]: Commit log is a bit cryptic (short), main reason for doing that is that the only method from localbinary.DriverPlugin which `RPCClientDriver.plugin` uses is `Close()`

File: libmachine/drivers/plugin/register_driver.go [hunk 4]
```diff
@@ -29,7 +29,7 @@ Please use this plugin through the main 'crc' binary.
 		os.Exit(1)
 	}
 
-	log.SetDebug(true)
+	log.SetLevel(log.DebugLevel)
 	os.Setenv("MACHINE_DEBUG", "1")
```
Comment (cfergeau) [inline libmachine/drivers/plugin/register_driver.go:33]: Unrelated, but nothing seems to be using `MACHINE_DEBUG`

**Review by cfergeau — APPROVED** [review]
(did not look much at  libmachine/drivers/plugin/localbinary/plugin_test.go )

**Review by gbraad — APPROVED** [review]
